src/core/server/plugins/types.ts

```diff
@@ -26,6 +26,23 @@ type Maybe<T> = T | undefined;
  */
 export type PluginConfigSchema<T> = Type<T>;
 
+/**
+ * Type defining the list of configuration properties that will be exposed on the client-side
+ * Object properties can either be fully exposed
+ *
+ * @public
+ */
+export type ExposedToBrowserDescriptor<T> = {
+  [Key in keyof T]?: T[Key] extends Maybe<any[]>
```

Comment:
nit: If we do `Array<infer U>` instead of `any[]` we remove the additional `any` and allow us to support cherry-picking properties in an array of objects if necessary 😇 

```js
const config = {
  myArray: [
    {
      propA: 'value',
      propB: 'value',
    }
  ]
};

const exposeToBrowser = {
  myArray: {
    propA: true
  }
}
```

Reply:
I 'stupidly' reproduced the type logic of `MakeUsageFromSchema` from the same file.

If we have a way to get rid of this `any`, I would be more than happy to. However regarding allowing to pick from the array, I'd rather hold that until we got a specific requirement for it, given the way we would be handling arrays in the `exposeToBrowser` definition would be slightly different and maybe a bit misleading (objectKey->properties, arrayKey->propertiesOfElement)